llvm/test/CodeGen/AMDGPU/llc-pipeline.ll

```diff
@@ -130,6 +130,8 @@
 ; GCN-O0-NEXT:        MachineDominator Tree Construction
 ; GCN-O0-NEXT:        Machine Natural Loop Construction
 ; GCN-O0-NEXT:        MachinePostDominator Tree Construction
+; GCN-O0-NEXT:        Basic Alias Analysis (stateless AA impl)
+; GCN-O0-NEXT:        Function Alias Analysis Results
```

Comment:
Can you skip querying for AA at -O0?

Reply:
If I just skip getAnalysis call it does not help since analysis is requested in the getAnalysisUsage. If I do not request it it is not available at any optlevel.  This is the benefit of the alternative https://github.com/llvm/llvm-project/pull/75974, it does not request the full analysis.

Reply:
You can check the opt level in getAnalysisUsage too, from the TargetMachine